### ISSUE TITLE
🦔 How can IMTD help us Load Test on Azure?

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#5554]
## ❓ Question

We know IMTD has offered to help us with load testing.  We also know Azure has some built-in tools for load testing.  What are the next steps?

## 🕵️ Details

How many simultaneous requests will our prod server be able to handle? (And how many should we expect per person, given all the requests for resource files and api calls per page.) Will our API queries slow to a crawl with many concurrent users?

## :question: Spike Questions
- Can we test on Dev or UAT and get results accurate for Prod?
- Will blasting our Azure server with requests incur costs?
- What are the next steps for load testing on Azure?

## 📦️ Timebox: 
1 Jira ticket

**Comment by brindasasi** [comment]
I will get better idea on this after #7999 
We are setting up self hosted vm and probably use that for this

**Comment by mnigh** [comment]
> I will get better idea on this after #7999 We are setting up self hosted vm and probably use that for this

@brindasasi any updates since then?

**Comment by brindasasi** [comment]
It wasnt the better solution as we can't maintain the vm by ourselves to keep the dependencies updated.
This question needs to be raised again. 
Please keep this ticket and I will get back to it.

**Comment by mnigh** [comment]
Has anyone looked into the Azure Load Testing service? It is even available for us? https://learn.microsoft.com/en-ca/azure/app-testing/load-testing/overview-what-is-azure-load-testing

**Comment by brindasasi** [comment]
I bump this because as part of scaling we need to know our current bench mark first .. Its pretty important right now

**Comment by brindasasi** [comment]
@tristan-orourke agreed to raise this on the scheduled IMTD meeting with Carl regarding the scaling.

**Comment by brindasasi** [comment]
so we talked about this .. a Jira ticket has been raised and I'm working with Eric Goodwin on this

**Comment by brindasasi** [comment]
> Has anyone looked into the Azure Load Testing service? It is even available for us? https://learn.microsoft.com/en-ca/azure/app-testing/load-testing/overview-what-is-azure-load-testing

this is what we are trying to do currently

**Comment by brindasasi** [comment]
- Can we test on Dev or UAT and get results accurate for Prod? 
   Yes. Currently configured for dev. Sample test ran for landing page 

<img width="2016" height="822" alt="Image" src="https://github.com/user-attachments/assets/7ee2b29e-7659-48e0-a2c2-113d30b06e92" />

   
- Will blasting our Azure server with requests incur costs?

**Cost Estimate**

Assuming:

Maximum of 2 App Service instances

- No Application Insights ingestion costs
- No serverless resources (Functions, Serverless SQL, etc.)
- No large file transfers

**Estimated incremental cost:**

- Small test (~40k requests): <$0.10 CAD
- Large test (~1 million requests): <$1-$2 CAD
- Conservative upper estimate for extended testing: <$5-$20 CAD


**What are the next steps for load testing on Azure?**
Need a test to bypass Canada Login and ability to test pages under authentication 
IMTD is willing to work with us on this

**Comment by brindasasi** [comment]
There was some config that IMTD made stopped load testing now and I'm no longer able to test even the landing pages.

<img width="2884" height="1828" alt="Image" src="https://github.com/user-attachments/assets/a2f65f27-07ed-4ccc-91cf-205b6e2e98fe" />

Reopened the jira ticket and waiting for the response now <insert any emoji you like here>

**Comment by brindasasi** [comment]
Latest follow up : Eric Goodwin is currently  working on getting this permission for me
